libraries/AP_HAL/utility/Socket.cpp

```diff
@@ -134,4 +162,30 @@ ssize_t SocketAPM::recv(void *buf, size_t size, uint32_t timeout_ms)
     return ::recv(fd, buf, size, 0);
 }
 
+ssize_t SocketAPM::recvfrom(void *buf, size_t size, const char *address, uint16_t port, uint32_t timeout)
```

Comment:
as I mentioned in another comment, this API makes no sense. A recvfrom() can't take an address as an input